### ISSUE TITLE
newly installed, rebinding keyboard controls for player one input stuck during UP

### ISSUE THREAD

**Issue by ghost** [alokforever/mh-reinstated#12]
I just downloaded the latest version today (3.2.2019) and attempted to play the game, but after loading a match I realized I forgot to bind the control first (singleplayer 1 keyboard only).

I hit the pause menu at the start of a match, went to controls, pressed the "config" button for player 1, keyboard controls and as it was requesting for me to press a button for "Up", I started trying to press the keys I wanted (WASD HJKL Space), it stopped recieveing input from the keyboard, even though during the intro it was responding to Enter to skip it.

I had to press control-alt-del to bring up the task manager and kill the program as nothing else was responding. Not esc, not alt-f4 and no other key on the keyboard.

TL;DR: Keyboard config input is broken. Requests UP key, but no input is recognized, leaving me stuck with an unresponsive program.

P.S. Additionally if I hit a lot of key on the keyboard while its requesting UP, the whole program goes blank and stops responding (Windows says this program has stopped responding)

**Comment by zeto0128** [comment]
Hi thanks for the report. This issue happened since a Windows update happened last November. This is probably a compatibility issue with Blitz 3D compiler and Windows 10 and we are still finding the solution to fix this problem. As a work around, you can configure the keys if you run the game using the Blitz 3D compiler and not the executable one. This, however, only works on keybord configuration as Joystick config has worse issues after the Windows update. 

You can download Blitz 3D here for free: https://blitzresearch.itch.io/blitz3d

After installing Blitz, open main.bb from it and press F5 to run. Hope it works for you.

**Comment by ghost** [comment]
Wait, I think it may be something different in my case.
* I'm using Windows 7 that has disabled all updates for the last 2 years.
* I have successfully played the 'master' branch edition of Multihero reinstated fine. (wolverine only causes crashes occasionally)
* Multihero 0.97 (thor) edition also works great with no issues nor crashes.

I'm currently having a lot of fun with the Thor edition so I may try the blitz3d trick much later. Thank you.

**Comment by zeto0128** [comment]
This brings me to the idea that the bug might be code-related issue (which is way better). However, what intrigues me is that the old base code (2017) that I used for the mod also already has the input problem. Not to mention that the input problem weren't there just last year. 

My plan is to read tutorials about the input configuration for blitz3d and create an empty blitz3d project to see if it has no issues. 

Thanks for the update.

**Comment by zeto0128** [comment]
Finally, I was able to implement a workaround against the keyboard configuration issue! Apparently, the nested Repeat loop was the underlying cause. I re-implemented the legacy code so that only one level of loop is used. The only issue left now is the Joystick detection.

**Comment by ghost** [comment]
Heck yeah! I'm going to give it a try and I'll let you know how it works on my end. I'm currently swarmed with IRL stuff, so I might not get to it in a few days.

**Comment by zeto0128** [comment]
Thanks man! Will be looking forward for the feedback.

**Comment by ghost** [comment]
I finally got an opportunity to try it and the issue is not gone. There is progress though.
I was able to define UP and DOWN keys, but then LEFT responded to nothing, so I had to ctrl-alt-del again.

**Comment by zeto0128** [comment]
I re-checked the code and I noticed that I accidentally commented out the code for flipping the display during Left key config. I'll test later after work.

**Comment by zeto0128** [comment]
Done.

**Comment by ghost** [comment]
Ok, I will test it out. thanks!

**Comment by ghost** [comment]
Wohoo! It works now, I was able to fully rebind the player 1 keyboard controls and play an entire match in battle mode. Success!
